src/uu/nl/src/helper.rs

```diff
@@ -107,16 +107,13 @@ pub fn parse_options(settings: &mut crate::Settings, opts: &clap::ArgMatches) ->
             }
         }
     }
-    match opts.get_one::<String>(options::NUMBER_WIDTH) {
-        None => {}
-        Some(val) => {
-            let conv: Option<usize> = val.parse().ok();
-            match conv {
-                None => {
-                    errs.push(String::from("Illegal value for -w"));
-                }
-                Some(num) => settings.number_width = num,
-            }
+    if let Some(num) = opts.get_one::<usize>(options::NUMBER_WIDTH) {
+        if *num > 0 {
+            settings.number_width = *num;
+        } else {
+            errs.push(String::from(
+                "Invalid line number field width: ‘0’: Numerical result out of range",
+            ));
```

Comment:
What about? Feels more rust idiomatic
```suggestion

    match opts.get_one::<usize>(options::NUMBER_WIDTH) {
        None => {},
        Some(num) if *num > 0 => settings.number_width = *num,
        Some(_) => errs.push(String::from(
            "Invalid line number field width: ‘0’: Numerical result out of range",
        )),
     }
```